### PR TITLE
Fix SSRF host canonicalization and add audit docs

### LINKED ISSUE #160: Track SSRF bypass from non-canonical loopback hosts
## Summary
Alternative numeric host spellings such as `2130706433`, `0x7f000001`, `0177.0.0.1`, and `127.1` can be normalized by the runtime to loopback addresses.

## Impact
This can bypass the intended SSRF protections when `allow_private_addresses` is `false` unless hosts are canonicalized before private-address checks.

## Expected Work
- Canonicalize hosts before direct lookup private-address checks
- Canonicalize redirect targets before redirect private-address checks
- Add regression tests for direct lookup and redirect cases

## Status
Fixed by PR #159.

### PR DESCRIPTION
**Pull request by silverbucket**
## Summary
Canonicalize hosts before SSRF checks so parser-normalized loopback forms like `2130706433`, `0x7f000001`, and `127.1` are blocked consistently in direct lookups and redirects.
Expand the Node/Bun regression suite and the browser bundle tests to cover direct lookup blocking, redirect blocking, and host normalization edge cases.
Closes #160.

### CODE REVIEW

File: src/webfinger.ts [hunk 1]
```diff
@@ -436,11 +462,26 @@ export default class WebFinger {
     }
 
     // Check for invalid characters that could indicate injection
-    if (cleanHost.includes('?') || cleanHost.includes('#') || cleanHost.includes(' ')) {
+    if (cleanHost.includes('?') || cleanHost.includes('#') || cleanHost.includes(' ') || cleanHost.includes('@')) {
```
Comment (raucao) [inline src/webfinger.ts]: How about something like this?

```ts
if (/[?# @]/.test(cleanHost)) {
  throw new WebFingerError('invalid characters in host');
}
```
